src/sidecar/sidecarHandle.ts

```diff
@@ -85,82 +87,59 @@ export class SidecarHandle {
-  ): T {
-    const config: Configuration = this.createClientConfig(configParams);
-    return new clientServiceClass(config);
-  }
-
-  // Return the ConnectionsResourceApi client instance for making REST requests to the sidecar.
-  public getConnectionsResourceApi(configParams?: ConfigurationParameters): ConnectionsResourceApi {
-    return this.getClient(ConnectionsResourceApi, configParams);
-  }
-
-  // Return the TemplatesApi client instance for making REST requests to the sidecar.
-  public getTemplatesApi(configParams?: ConfigurationParameters): TemplatesApi {
-    return this.getClient(TemplatesApi, configParams);
-  }
-
-  public VersionResourceApi(configParams?: ConfigurationParameters): VersionResourceApi {
-    return this.getClient(VersionResourceApi, configParams);
-  }
+  // --- SIDECAR OPENAPI CLIENT METHODS ---
 
-  public getTopicV3Api(clusterId: string, connectionId: string): TopicV3Api {
-    const config: unknown = this.createClientConfig({
-      headers: { "x-cluster-id": clusterId, "x-connection-id": connectionId },
-    });
-    return new TopicV3Api(config as KafkaRestConfiguration);
+  /**
+   * Creates and returns a (Sidecar REST OpenAPI spec) {@link ConnectionsResourceApi} client instance with a preconfigured
+   * {@link SidecarRestConfiguration}.
+   */
+  public getConnectionsResourceApi(): ConnectionsResourceApi {
+    const config = new SidecarRestConfiguration(this.defaultClientConfigParams);
+    return new ConnectionsResourceApi(config);
   }
 
-  public getPartitionV3Api(clusterId: string, connectionId: string): PartitionV3Api {
-    const config: unknown = this.createClientConfig({
-      headers: { "x-cluster-id": clusterId, "x-connection-id": connectionId },
+  /**
+   * Creates and returns a (Sidecar REST OpenAPI spec) {@link TemplatesApi} client instance with a
+   * preconfigured {@link SidecarRestConfiguration}.
+   */
+  public getTemplatesApi(): TemplatesApi {
+    const config = new SidecarRestConfiguration({
+      ...this.defaultClientConfigParams,
+      headers: {
+        ...this.defaultClientConfigParams.headers,
+        // Intercept requests to set the Accept header to `application/*` to handle non-JSON responses
+        // For example, the sidecar returns a ZIP file for the `POST /gateway/v1/templates/{name}/apply` endpoint
+        Accept: "application/*",
+      },
     });
-    return new PartitionV3Api(config as KafkaRestConfiguration);
+    return new TemplatesApi(config);
   }
 
-  public getSchemasV1Api(clusterId: string, connectionId: string): SchemasV1Api {
-    const config: unknown = this.createClientConfig({
-      headers: { "x-cluster-id": clusterId, "x-connection-id": connectionId },
+  /**
+   * Creates and returns a (Sidecar REST OpenAPI spec) {@link VersionResourceApi} client instance
+   * with a preconfigured {@link SidecarRestConfiguration}.
+   */
+  public getVersionResourceApi(configParams?: ConfigurationParameters): VersionResourceApi {
```

Comment:
Renamed from `VersionResourceApi`, although I don't have any strong feelings for/against the `get<class>` vs just `<class>` naming. This was updated to match the rest of these get-client-instance-from-method naming patterns, but I'm good with moving to just the class name if folks would rather use that style instead.